packages/mui-utils/src/elementAcceptingRef.test.tsx

```diff
@@ -26,6 +26,8 @@ describe('elementAcceptingRef', () => {
       function testAct() {
         checkPropType(element);
         if (shouldMount) {
+          // TODO: replace with React 18 implementation after https://github.com/testing-library/react-testing-library/issues/1216 is closed.
```

Comment:
Purely out of curiosity, did you try it with https://github.com/testing-library/react-testing-library/issues/1216#issuecomment-1577902646?